add stations support to nearby view

daniel-heppner-ibigroup opened the pull request:
<!--Please provide a brief description of what this PR accomplishes and note if it should be considered/merged with any related PR(s)-->
**Description:**
Adds support for Stations with sub-stops in nearby view. This version is based on dev instead of the scooter merging.

danielhep committed:
add stations support to nearby view

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -19,12 +20,21 @@ const { getUserTimezone } = coreUtils.time
 
+// Style for child stop headers
+const ChildStopHeader = styled.div`
+  background-color: #f5f5f5;
```
Why is the color gray? Can it be transparent/white instead?

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -19,12 +20,21 @@ const { getUserTimezone } = coreUtils.time
+  background-color: #f5f5f5;
+  border-radius: 4px 4px 0 0;
+  font-size: 14px;
+  font-weight: bold;
```
This doesn't look great. Can we do hierarchy with just font size?

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -19,12 +20,21 @@ const { getUserTimezone } = coreUtils.time
+  border-radius: 4px 4px 0 0;
+  font-size: 14px;
+  font-weight: bold;
+  padding: 10px 15px;
```
Can we have the padding match the units we're using elsewhere? I think we're on units of 8

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -95,20 +105,40 @@ const Stop = ({
         alwaysShowLongName={nearbyViewConfig?.alwaysShowLongName}
         homeTimezone={homeTimezone}
         key={index}
+        noRoundedBottom={noRoundedBottom}
```
Any way we can avoid a prop and use css instead?

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -95,20 +105,40 @@ const Stop = ({
     </StyledAlert>
   )
 
+  console.log(patternRows)
```
oopsie

miles-grant-ibigroup commented on lib/components/viewers/styled.ts:
```diff
@@ -145,8 +148,10 @@ export const PatternRowItem = styled.li<{ roundedTop?: boolean }>`
   }
 
   &:last-of-type .header.stop-view {
```
Can we not take advantage of selectors instead of adding a new prop? Would make a lot of this code much much cleaner

miles-grant-ibigroup reviewed (changes requested):
Only a few small cleanups needed! Mostly in getting rid of that new prop and making the new headers less bold. Can we have it match the From Here/To Here? 

Also "view schedule" doesn't work

danielhep committed:
address PR comments
- remove rounded corners on pattern rows
- hide parent schedule if there are child stops
- improve styling/shadows
- move view schedule button to child stop headers for stations

daniel-heppner-ibigroup commented:
I think I made some big improvements in the latest commit, let me know what you think. I'm still not sure about the styling for the headers.

danielhep committed:
make child stop text a little bigger

danielhep committed:
update snapshot

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop-card-header.tsx:
```diff
@@ -78,14 +78,16 @@ const StopCardHeader = ({
+          {actionPath && actionText ? (
+            <Link
+              className="pull-right"
+              style={{ color: 'inherit', fontSize: 'small' }}
```
Any way we can get some hover underlines on this?

daniel-heppner-ibigroup replied:
<img width="136" alt="image" src="https://github.com/user-attachments/assets/60ac7fbd-5538-4e05-94fb-c9a97f46b074" />
??

daniel-heppner-ibigroup replied:
I moved it to a styled component anyway

miles-grant-ibigroup replied:
Probably a safari issue

miles-grant-ibigroup commented on lib/components/viewers/pattern-row.tsx:
```diff
@@ -25,6 +25,7 @@ import StopTimeCell from './stop-time-cell'
 type Props = {
   alwaysShowLongName?: boolean
   homeTimezone?: string
+  noRoundedBottom?: boolean
```
I think this can be dropped now

miles-grant-ibigroup replied:
It's still here

miles-grant-ibigroup commented on lib/components/viewers/pattern-row.tsx:
```diff
@@ -59,11 +60,11 @@ const renderDay = (homeTimezone: string, day: number): JSX.Element => {
  * Represents a single pattern row for displaying arrival times in the stop
  * viewer.
  */
+// eslint-disable-next-line complexity
```
I think this can be dropped now

daniel-heppner-ibigroup replied:
still needed

miles-grant-ibigroup replied:
Are you sure

daniel-heppner-ibigroup replied:
<img width="716" alt="image" src="https://github.com/user-attachments/assets/84854a58-ca39-458a-979b-34024974b3c5" />

miles-grant-ibigroup reviewed (changes requested):
Great work with the CSS! I'm in agreement something is off with the headers now. Maybe same font size but higher font weight? Otherwise code looks pretty good and working well for me. Will leave it up to the second reviewer to nitpick spacings and paddings

danielhep committed:
remove unused props

danielhep committed:
move action link to styled component

danielhep committed:
re-add the bolt text for @miles-grant-ibi

miles-grant-ibigroup reviewed (approved):
Beautiful! Still a few rough edges but happy for this to go to second review. Snapshots seem messed up you might want to clear your jest cache.

danielhep committed:
update snapshots and fix percy

amy-corson-ibigroup committed:
Merge branch 'dev' into nearby-view-stations-basedev

amy-corson-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -3,28 +3,44 @@ import { connect } from 'react-redux'
   (stoptime.serviceDay || 0) + (stoptime.realtimeDeparture || 0)
 
+// Style for child stop headers
+const ChildStopHeader = styled.div`
```
Can we add an `align-items: center` here? The header and link are looking a little misaligned 
![image](https://github.com/user-attachments/assets/c3556a1c-5224-4124-a6a3-ab1e5770b1b3)

amy-corson-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -3,28 +3,44 @@ import { connect } from 'react-redux'
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 8px 16px;
```
I think `10px 16px` could make these labels look a little less cramped!

amy-corson-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -96,35 +111,90 @@ const Stop = ({
         stopData={stopData}
       />
       <div>
         <div>{timezoneWarning}</div>
```
Could we add a `margin-bottom: 1rem` on the departure timezone notice? I think it could help the overall spacing (not just for stations) because it matches the margin between it and the from/to picker.
| Station | Stops|
|--------|-------|
|![image](https://github.com/user-attachments/assets/84cf7ab4-988e-49dc-9496-66dabed6fb75)|![image](https://github.com/user-attachments/assets/975974fc-1ce0-4d72-a9a5-29adc0960362)|

daniel-heppner-ibigroup replied:
I feel like that is too much whitespace tbh, not a fan. Especially because it's repeated on each card, it is a lot of extra space for something that isn't that important.

amy-corson-ibigroup reviewed (changes requested):
This looks great, thank you so much for the changes! How would we feel about upping the top/bottom padding on the label, increasing the margin between the first label and the "departure times" notice, and adding either a border or a background color? I think the first stop label is bleeding into the station card, adding a little separation could make it a little clearer: 
| Now | Border | Background | 
|--------|-------|-------|
|![image](https://github.com/user-attachments/assets/9f7ef797-01b3-48e8-8cae-ad38ce777eb1)|![image](https://github.com/user-attachments/assets/8450457e-475a-4770-9292-76a02ef644bd)|![image](https://github.com/user-attachments/assets/580e59c6-665b-42a3-b929-3e39211736dd)

(Requesting changes for the convo but nothing technically blocking here, so if we decide to keep as is I'll approve)

miles-grant-ibigroup commented:
> This looks great, thank you so much for the changes! How would we feel about upping the top/bottom padding on the label, increasing the margin between the first label and the "departure times" notice, and adding either a border or a background color? I think the first stop label is bleeding into the station card, adding a little separation could make it a little clearer:
> 
> Now	Border	Background
> ![image](https://private-user-images.githubusercontent.com/115499534/429057523-9f7ef797-01b3-48e8-8cae-ad38ce777eb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM1MTgxNDAsIm5iZiI6MTc0MzUxNzg0MCwicGF0aCI6Ii8xMTU0OTk1MzQvNDI5MDU3NTIzLTlmN2VmNzk3LTAxYjMtNDhlOC04Y2FlLWFkMzhjZTc3N2ViMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwMVQxNDMwNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MGFlNWUxMWJiNjU3NTljZjRjMThjZDg5ZjNkZDJkMTViZjlmOGU1NWRhNTQ5MzYzNGNiNzYxNGEzYTU4YTY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Bm2C125MgBT48RcN122Y0zpWFq1fuj104Snv5O_OtKY)	![image](https://private-user-images.githubusercontent.com/115499534/429057415-8450457e-475a-4770-9292-76a02ef644bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM1MTgxNDAsIm5iZiI6MTc0MzUxNzg0MCwicGF0aCI6Ii8xMTU0OTk1MzQvNDI5MDU3NDE1LTg0NTA0NTdlLTQ3NWEtNDc3MC05MjkyLTc2YTAyZWY2NDRiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwMVQxNDMwNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZGNiNzg3YjJhZWRmYjNlMWY2MzIwNGE1ZjM1ZmQ5MmI2YjMyOWM3ODE4ZmExYWUyN2M0N2YxMzEwMWZiOWNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ljLtQ2xWlK2WJpp0zsmlPb3_TSaOTDsnqs72sMq5q-E)	![image](https://private-user-images.githubusercontent.com/115499534/429058506-580e59c6-665b-42a3-b929-3e39211736dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM1MTgxNDAsIm5iZiI6MTc0MzUxNzg0MCwicGF0aCI6Ii8xMTU0OTk1MzQvNDI5MDU4NTA2LTU4MGU1OWM2LTY2NWItNDJhMy1iOTI5LTNlMzkyMTE3MzZkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwMVQxNDMwNDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NTU2YzQxMjdhOGZkYmM2ZGZmMjFiOGIwYmQzZTI0OTY4YWFiMDBkMWZkNmYxNTdjNTYzZWY0MjE1MjEzN2ZmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pTv2u4fIBvb6qYfHXqzFWdQBxDJ2r2UjXoxDMVHVghE)
> (Requesting changes for the convo but nothing technically blocking here, so if we decide to keep as is I'll approve)

I love the bigger border and the centering! The gray background I have some reservations about (maybe something transparent instead?) but I think that should wait for a future PR!

amy-corson-ibigroup committed:
Merge branch 'dev' into nearby-view-stations-basedev

daniel-heppner-ibigroup committed:
Merge branch 'dev' into nearby-view-stations-basedev

danielhep committed:
update design based on feedback

danielhep committed:
Merge branch 'nearby-view-stations-basedev' of github.com:opentripplanner/otp-react-redux into nearby-view-stations-basedev

daniel-heppner-ibigroup commented:
I liked the border proposal more too, so I added a thin border to the top. Let me know if it's good! 
![image](https://github.com/user-attachments/assets/70a72d63-f20a-47b3-b1ae-1f76717d775f)

danielhep committed:
update snapshots

amy-corson-ibigroup reviewed (approved):
Looks great! Thanks for the changes!!

amy-corson-ibigroup committed:
Merge branch 'dev' into nearby-view-stations-basedev

miles-grant-ibigroup commented on lib/components/viewers/nearby/stop.tsx:
```diff
@@ -25,14 +25,19 @@ export const fullTimestamp = (stoptime: StopTime) =>
   }
+
+  &:first-of-type {
+    border-top: 1px solid #999999;
```
Could we use the nearby view background color instead?

miles-grant-ibigroup reviewed (changes requested):
I'm happy with the border compromise! Just one small color note

danielhep committed:
use main base color in border

danielhep committed:
Merge branch 'nearby-view-stations-basedev' of github.com:opentripplanner/otp-react-redux into nearby-view-stations-basedev

miles-grant-ibigroup reviewed (approved):
What do you think about decreasing the transparency of the line?

danielhep committed:
transparent border

daniel-heppner-ibigroup committed:
Merge branch 'dev' into nearby-view-stations-basedev
